go.mod

```diff
@@ -1,8 +1,10 @@
 module github.com/ionos-cloud/cluster-api-provider-proxmox
 
-go 1.21
+go 1.22.0
```

Comment:
It's better to set `go 1.22.2` and remove toolchain

Reply:
indeed, pushing the change now.